Regression with Zarr: ReadOnlyError

Tests with the latest dev environment are failing with errors like this

```

tmp_path = PosixPath('/private/var/folders/kl/7rfdrpx96bb0rhbnl5l2dnkw0000gn/T/pytest-of-rabernat/pytest-69/test_rechunk_group_mapper_temp7')
executor = 'python', source_store = 'mapper.source.zarr', target_store = <fsspec.mapping.FSMap object at 0x1174e3520>
temp_store = <fsspec.mapping.FSMap object at 0x1174e3400>

    @pytest.mark.parametrize(
        "executor",
        [
            "dask",
            "python",
            requires_beam("beam"),
            requires_prefect("prefect"),
        ],
    )
    @pytest.mark.parametrize("source_store", ["source.zarr", "mapper.source.zarr"])
    @pytest.mark.parametrize("target_store", ["target.zarr", "mapper.target.zarr"])
    @pytest.mark.parametrize("temp_store", ["temp.zarr", "mapper.temp.zarr"])
    def test_rechunk_group(tmp_path, executor, source_store, target_store, temp_store):
        if source_store.startswith("mapper"):
            fsspec = pytest.importorskip("fsspec")
            store_source = fsspec.get_mapper(str(tmp_path) + source_store)
            target_store = fsspec.get_mapper(str(tmp_path) + target_store)
            temp_store = fsspec.get_mapper(str(tmp_path) + temp_store)
        else:
            store_source = str(tmp_path / source_store)
            target_store = str(tmp_path / target_store)
            temp_store = str(tmp_path / temp_store)
    
>       group = zarr.group(store_source)

tests/test_rechunk.py:457: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../../mambaforge/envs/rechunker/lib/python3.9/site-packages/zarr/hierarchy.py:1355: in group
    init_group(store, overwrite=overwrite, chunk_store=chunk_store,
../../../mambaforge/envs/rechunker/lib/python3.9/site-packages/zarr/storage.py:648: in init_group
    _init_group_metadata(store=store, overwrite=overwrite, path=path,
../../../mambaforge/envs/rechunker/lib/python3.9/site-packages/zarr/storage.py:711: in _init_group_metadata
    store[key] = store._metadata_class.encode_group_metadata(meta)  # type: ignore
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <zarr.storage.FSStore object at 0x1174e34c0>, key = '.zgroup', value = b'{\n    "zarr_format": 2\n}'

    def __setitem__(self, key, value):
        if self.mode == 'r':
>           raise ReadOnlyError()
E           zarr.errors.ReadOnlyError: object is read-only

../../../mambaforge/envs/rechunker/lib/python3.9/site-packages/zarr/storage.py:1410: ReadOnlyError
==
```

This is the cause of the test failures in #134.

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -1,6 +1,6 @@
 repos:
 -   repo: https://github.com/pycqa/isort
-    rev: 5.6.4
+    rev: 5.12.0
     hooks:
       - id: isort
         args: ["--profile", "black", "--filter-files"]
@@ -9,7 +9,7 @@ repos:
     hooks:
     - id: black
       language_version: python3
--   repo: https://gitlab.com/pycqa/flake8
+-   repo: https://github.com/pycqa/flake8
     rev: 3.7.9
     hooks:
     - id: flake8

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -15,6 +15,6 @@ repos:
     - id: flake8
       language_version: python3
 -   repo: https://github.com/pre-commit/mirrors-mypy
-    rev: v0.790
+    rev: v1.0.1
     hooks:
     -   id: mypy

--- a/rechunker/pipeline.py
+++ b/rechunker/pipeline.py
@@ -1,6 +1,6 @@
 import itertools
 import math
-from typing import Any, Iterable, Iterator, Tuple, TypeVar
+from typing import Any, Iterable, Iterator, Tuple
 
 import dask
 import numpy as np
@@ -74,11 +74,8 @@ def specs_to_pipelines(specs: Iterable[CopySpec]) -> ParallelPipelines:
     return tuple((spec_to_pipeline(spec) for spec in specs))
 
 
-T = TypeVar("T")
-
-
 class CopySpecToPipelinesMixin(CopySpecExecutor):
-    def prepare_plan(self, specs: Iterable[CopySpec]) -> T:
+    def prepare_plan(self, specs: Iterable[CopySpec]):
         pipelines = specs_to_pipelines(specs)
         return self.pipelines_to_plan(pipelines)
 

--- a/tests/test_rechunk.py
+++ b/tests/test_rechunk.py
@@ -209,7 +209,7 @@ def test_rechunk_dataset(
     xarray = pytest.importorskip("xarray")
 
     if target_store.startswith("mapper"):
-        fsspec = pytest.importorskip("fsspec")
+        pytest.importorskip("fsspec")
         target_store = FSStore(str(tmp_path) + target_store)
         temp_store = FSStore(str(tmp_path) + temp_store)
     else:
@@ -446,7 +446,7 @@ def test_rechunk_dask_array(
 @pytest.mark.parametrize("temp_store", ["temp.zarr", "mapper.temp.zarr"])
 def test_rechunk_group(tmp_path, executor, source_store, target_store, temp_store):
     if source_store.startswith("mapper"):
-        fsspec = pytest.importorskip("fsspec")
+        pytest.importorskip("fsspec")
         store_source = FSStore(str(tmp_path) + source_store)
         target_store = FSStore(str(tmp_path) + target_store)
         temp_store = FSStore(str(tmp_path) + temp_store)

--- a/tests/test_rechunk.py
+++ b/tests/test_rechunk.py
@@ -9,6 +9,7 @@ import numpy
 import numpy as np
 import pytest
 import zarr
+from zarr.storage import FSStore
 
 from rechunker import api
 
@@ -209,8 +210,8 @@ def test_rechunk_dataset(
 
     if target_store.startswith("mapper"):
         fsspec = pytest.importorskip("fsspec")
-        target_store = fsspec.get_mapper(str(tmp_path) + target_store)
-        temp_store = fsspec.get_mapper(str(tmp_path) + temp_store)
+        target_store = FSStore(str(tmp_path) + target_store)
+        temp_store = FSStore(str(tmp_path) + temp_store)
     else:
         target_store = str(tmp_path / target_store)
         temp_store = str(tmp_path / temp_store)
@@ -446,15 +447,15 @@ def test_rechunk_dask_array(
 def test_rechunk_group(tmp_path, executor, source_store, target_store, temp_store):
     if source_store.startswith("mapper"):
         fsspec = pytest.importorskip("fsspec")
-        store_source = fsspec.get_mapper(str(tmp_path) + source_store)
-        target_store = fsspec.get_mapper(str(tmp_path) + target_store)
-        temp_store = fsspec.get_mapper(str(tmp_path) + temp_store)
+        store_source = FSStore(str(tmp_path) + source_store)
+        target_store = FSStore(str(tmp_path) + target_store)
+        temp_store = FSStore(str(tmp_path) + temp_store)
     else:
         store_source = str(tmp_path / source_store)
         target_store = str(tmp_path / target_store)
         temp_store = str(tmp_path / temp_store)
 
-    group = zarr.group(store_source)
+    group = zarr.group(store_source, overwrite=True)
     group.attrs["foo"] = "bar"
     # 800 byte chunks
     a = group.ones("a", shape=(5, 10, 20), chunks=(1, 10, 20), dtype="f4")

